### PR TITLE
[MM-55330]: Convert `./components/widgets/popover/index.tsx` from Class Component to Function Component 

### PR DESCRIPTION
**Pull request by 0xJamin**
#### Summary

This PR converts the popover widgets from a class component to a function component and also updates the snapshots of other components that use the popover widgets.
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/25359
Jira https://mattermost.atlassian.net/browse/MM-55330

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/widgets/popover/popover.test.tsx [hunk 1]
```diff
@@ -1,18 +1,18 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {shallow} from 'enzyme';
+import {render} from '@testing-library/react';
```
Comment (0xJamin) [inline webapp/channels/src/components/widgets/popover/popover.test.tsx:4]: Also, I converted the test to use `@testing-library`

Comment (0xJamin) [reply webapp/channels/src/components/widgets/popover/popover.test.tsx:4]: /check-cla

File: webapp/channels/src/components/widgets/popover/index.tsx [hunk 2]
```diff
@@ -21,30 +21,23 @@ interface Props {
     onMouseOver?: React.MouseEventHandler<BSPopover>;
 }
 
-export default class Popover extends React.PureComponent<Props> {
-    static defaultProps = {
-        placement: 'right',
-        popoverStyle: 'info',
-        popoverSize: 'sm',
+const Popover = ({placement = 'right', popoverSize = 'sm', children, popoverStyle = 'info', title, id, onMouseOut, onMouseOver, className, style}: Props) => {
```
Comment (larkox) [inline webapp/channels/src/components/widgets/popover/index.tsx:24]: To improve readability, keep one Prop per line:

```suggestion
const Popover = ({
    placement = 'right',
    popoverSize = 'sm',
    children,
    popoverStyle = 'info',
    title,
    id,
    onMouseOut,
    onMouseOver,
    className,
    style,
}: Props) => {
```

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just one minor change and looks fine :smile:

**Comment by 0xJamin** [comment]
@larkox  changes updated

**Review by larkox — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]
Amazing work thank you 👍🏽

**Comment by larkox** [comment]
@0xJamin There seems to be some type errors. Could you take a look?

**Comment by larkox** [comment]
@0xJamin There seems to be some errors in the PR (regarding types). Can you look at those?

**Comment by 0xJamin** [comment]
Hi @larkox I'll check it out today

**Comment by 0xJamin** [comment]
Hi @larkox, Can you run the tests now to see if it passes?

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**